### ISSUE TITLE
Intersector calculates a wrong result with some triangles missing.

### ISSUE THREAD

**Issue by nilsschmidt1337** [nilsschmidt1337/ldparteditor#956]
Here is a minimal testcase:
```
0 Testcase
0 Name: new.dat
0 Author: Nils Schmidt [BlackBrick89]
0 !LDRAW_ORG Unofficial_Part
0 !LICENSE Licensed under CC BY 2.0 and CC BY 4.0 : see CAreadme.txt

0 BFC CERTIFY CCW

1 16 0 0 2 .90613 .19509 -.37533 -.38268 0 -.92388 -.18024 .98079 .07466 stud.dat
4 1 0 10 0 10 10 0 10 0 0 0 0 0

```

Expected result (with the selection of **4 1 0 10 0 10 10 0 10 0 0 0 0 0**):

```
1 16 0 0 2 .90613 .19509 -.37533 -.38268 0 -.92388 -.18024 .98079 .07466 stud.dat

3 1 0 10 0 10 10 0 10 0 0
3 1 0 10 0 10 0 0 3.928084 4.242727 0
3 1 0 5.922502 0 0 10 0 1.943262 5.54328 0
3 1 0 0 0 0 5.922502 0 1.943262 5.54328 0
3 1 0 10 0 3.928084 4.242727 0 1.943262 5.54328 0
3 1 0 0 0 1.943262 5.54328 0 3.928084 4.242727 0
3 1 3.928084 4.242727 0 10 0 0 5.25401 2.296095 0
3 1 0 0 0 3.928084 4.242727 0 5.25401 2.296095 0
3 1 5.719879 0 0 0 0 0 5.25401 2.296095 0
3 1 5.719879 0 0 5.25401 2.296095 0 10 0 0

```

Minimalistic error case:
```
4 16 4.656362196 2.29609512 -3.00459874 5.436722196 2.29609512 .91856126 5.884873788 .000064944 .829423292 5.104513788 .000064944 -3.093736708
4 16 5.104513788 .000064944 -3.093736708 5.884873788 .000064944 .829423292 5.43678 -2.29608 .91856 4.65642 -2.29608 -3.0046

4 1 0 10 0 10 10 0 10 0 0 0 0 0

```

Error case with just three triangles (easy to debug):
```
3 16 5.884873788 .000064944 .829423292 5.104513788 .000064944 -3.093736708 4.656362196 2.29609512 -3.00459874
3 16 5.104513788 .000064944 -3.093736708 5.884873788 .000064944 .829423292 5.43678 -2.29608 .91856

3 1 10 0 0 0 0 0 0 10 0

```

**Comment by nilsschmidt1337** [comment]
Another complex testcase:
```
0 Testcase
0 Name: new.dat
0 Author: Nils Schmidt [BlackBrick89]
0 !LDRAW_ORG Unofficial_Part
0 !LICENSE Licensed under CC BY 2.0 and CC BY 4.0 : see CAreadme.txt

0 BFC CERTIFY CCW

4 41 11.606700308265506855152363865737 .1675675358136919364100801349153 -8.0781510505743379621308579488637 -8.0042710612078028417163289239998 2.1683236515759270635993006787866 -11.45557123312383866456924832022 -11.606700308265506855152363865737 -.1675675358136919364100801349153 8.0781510505743379621308579488637 8.0042710612078028417163289239998 -2.1683236515759270635993006787866 11.45557123312383866456924832022
4 7 1.75 2 6 1.75 -2 6 5.9926 -2 4.2426 5.9926 2 4.2426

0 // Select this quad and run intersector:
4 2 -2.4926 2 4.2426 -2.4926 -2 4.2426 1.75 -2 6 1.75 2 6

4 7 -4.25 2 0 -4.25 -2 0 -2.4926 -2 4.2426 -2.4926 2 4.2426
3 7 1.75 -2 0 1.75 -2 6 -2.4926 -2 4.2426

1 7 -1.75 2 0 1 0 0 0 1 0 0 0 1 stu2.dat

```

**Comment by nilsschmidt1337** [comment]
Closed with commit [5060395](https://github.com/nilsschmidt1337/ldparteditor/commit/506039583cbe3df8b5a32eb210555a19723c1154).

**Comment by nilsschmidt1337** [comment]
Implementation was incomplete.

**Comment by nilsschmidt1337** [comment]
Closed with commit [7b0a67e](https://github.com/nilsschmidt1337/ldparteditor/commit/7b0a67e9d8c6e296150ca92f192cceeaa13a09b9).
